// enkanetwork/dist/models/characters/characterSkill.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CharacterSkill = void 0;
const getAssetUrl_1 = require("../../helpers/getAssetUrl");
const getJsonAssets_1 = require("../../helpers/getJsonAssets");
class CharacterSkill {
    constructor(lang, skill, level, proudSkillExtraLevelMap) {
        const charactersSkillsAsset = getJsonAssets_1.charactersSkillsAssets[skill];
        const charactersSkillsLocalization = getJsonAssets_1.charactersSkillsLocalizations[charactersSkillsAsset.nameTextMapHash];
        const boost = typeof charactersSkillsAsset.proudSkillGroupId == "number"
            ? Object.keys(proudSkillExtraLevelMap).find((x) => x == String(charactersSkillsAsset.proudSkillGroupId))
            : false;
        this.id = skill;
        this.icon = (0, getAssetUrl_1.getAssetUrl)(charactersSkillsAsset.skillIcon);
        this.name = charactersSkillsLocalization[lang];
        this.isBoosted = !!boost;
        this.level = level + (boost ? +proudSkillExtraLevelMap[boost] : 0); //my code has gotten worse lately...
    }
}
exports.CharacterSkill = CharacterSkill;
