// enkanetwork/dist/models/characters/characterReluquary.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CharacterReluquary = void 0;
const getAssetUrl_1 = require("../../helpers/getAssetUrl");
const getJsonAssets_1 = require("../../helpers/getJsonAssets");
const reluquaryTypes = {
    EQUIP_BRACER: "Flower",
    EQUIP_NECKLACE: "Feather",
    EQUIP_SHOES: "Sands",
    EQUIP_RING: "Goblet",
    EQUIP_DRESS: "Circlet",
};
class CharacterReluquary {
    constructor(lang, equipment) {
        this.id = equipment.itemId;
        this.name =
            getJsonAssets_1.charactersReluquaryLocalizations[equipment.flat.nameTextMapHash][lang];
        this.setName =
            getJsonAssets_1.charactersReluquarySetsLocalizations[equipment.flat.setNameTextMapHash][lang];
        this.icon = (0, getAssetUrl_1.getAssetUrl)(equipment.flat.icon);
        this.type = reluquaryTypes[equipment.flat.equipType];
        this.level = --equipment.reliquary.level;
        this.rarity = equipment.flat.rankLevel;
        this.mainStats = equipment.flat.reliquaryMainstat;
        this.subStats = equipment.flat.reliquarySubstats;
    }
}
exports.CharacterReluquary = CharacterReluquary;
