// enkanetwork/dist/models/characters/characterWeapon.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CharacterWeapon = void 0;
const getAssetUrl_1 = require("../../helpers/getAssetUrl");
const getJsonAssets_1 = require("../../helpers/getJsonAssets");
class CharacterWeapon {
    constructor(lang, equipment) {
        this.id = equipment.itemId;
        this.name =
            getJsonAssets_1.charactersWeaponLocalizations[equipment.flat.nameTextMapHash][lang];
        this.icon = (0, getAssetUrl_1.getAssetUrl)(equipment.flat.icon);
        this.level = equipment.weapon.level;
        this.elevations = equipment.weapon.promoteLevel || 0;
        this.improvement = equipment.weapon.affixMap
            ? equipment.weapon.affixMap[Object.keys(equipment.weapon.affixMap)[0]] + 1
            : 1;
        this.rarity = equipment.flat.rankLevel;
        this.mainStat = equipment.flat.weaponStats[0];
        this.subStat = equipment.flat.weaponStats[1] || false;
    }
}
exports.CharacterWeapon = CharacterWeapon;
