// enkanetwork/dist/helpers/getJsonAssets.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.namecardsLocalizations = exports.namecardsAssets = exports.charactersReluquarySetsLocalizations = exports.charactersReluquaryLocalizations = exports.charactersWeaponLocalizations = exports.charactersSkillsLocalizations = exports.charactersSkillsAssets = exports.charactersConstellationLocalizations = exports.charactersConstellationAssets = exports.charactersLocalizations = exports.charactersCostumes = exports.charactersAssets = void 0;
// @ts-ignore: Json Import
const characters_json_1 = __importDefault(require("../../assets/data/characters.json"));
exports.charactersAssets = characters_json_1.default;
// @ts-ignore: Json Import
const costumes_json_1 = __importDefault(require("../../assets/data/costumes.json"));
exports.charactersCostumes = costumes_json_1.default;
// @ts-ignore: Json Import
const characters_json_2 = __importDefault(require("../../assets/localizations/characters.json"));
exports.charactersLocalizations = characters_json_2.default;
// @ts-ignore: Json Import
const constellations_json_1 = __importDefault(require("../../assets/data/constellations.json"));
exports.charactersConstellationAssets = constellations_json_1.default;
// @ts-ignore: Json Import
const constellations_json_2 = __importDefault(require("../../assets/localizations/constellations.json"));
exports.charactersConstellationLocalizations = constellations_json_2.default;
// @ts-ignore: Json Import
const skills_json_1 = __importDefault(require("../../assets/data/skills.json"));
exports.charactersSkillsAssets = skills_json_1.default;
// @ts-ignore: Json Import
const skills_json_2 = __importDefault(require("../../assets/localizations/skills.json"));
exports.charactersSkillsLocalizations = skills_json_2.default;
// @ts-ignore: Json Import
const weapons_json_1 = __importDefault(require("../../assets/localizations/weapons.json"));
exports.charactersWeaponLocalizations = weapons_json_1.default;
// @ts-ignore: Json Import
const artifacts_json_1 = __importDefault(require("../../assets/localizations/artifacts.json"));
exports.charactersReluquaryLocalizations = artifacts_json_1.default;
// @ts-ignore: Json Import
const artifactSets_json_1 = __importDefault(require("../../assets/localizations/artifactSets.json"));
exports.charactersReluquarySetsLocalizations = artifactSets_json_1.default;
// @ts-ignore: Json Import
const namecards_json_1 = __importDefault(require("../../assets/data/namecards.json"));
exports.namecardsAssets = namecards_json_1.default;
// @ts-ignore: Json Import
const namecards_json_2 = __importDefault(require("../../assets/localizations/namecards.json"));
exports.namecardsLocalizations = namecards_json_2.default;
